syft/source/source.go

```diff
@@ -353,27 +355,28 @@ func (s *Source) SetID() {
 	}
 
 	s.id = artifact.ID(strings.TrimPrefix(d, "sha256:"))
+	s.Metadata.ID = strings.TrimPrefix(d, "sha256:")
 }
 
-func calculateChainID(img *image.Image) string {
-	if len(img.Layers) < 1 {
+func calculateChainID(m ImageMetadata) string {
```

Comment:
nit: it looks like this doens't need the full metadata, could we just pass the layers to keep the concerns smaller for this function?